pkg/apis/work/v1alpha2/binding_types.go

```diff
@@ -347,6 +357,8 @@ type ResourceBindingStatus struct {
 
 	// Conditions contain the different condition statuses.
 	// +optional
+	// +listType=map
+	// +listMapKey=type
```

Comment:
Why add these? What does it mean?

Reply:
It will be useful to manage conditions as a map. but current PR doesn't require it.

Reply:
Yeah we can track it at https://github.com/karmada-io/karmada/pull/6002, which is needed by `ApplyStatus` method.

Reply:
removed.